### PR TITLE
SWORD2: Add Hebrew support

### PR DESCRIPTION
**Pull request by orgads**
CC @ShaharAriel

### CODE REVIEW

**Comment by orgads** [comment]
Please do not merge yet. It's incomplete.

File: engines/sword2/maketext.cpp [hunk 1]
```diff
@@ -266,17 +267,22 @@ byte *FontRenderer::buildTextSprite(const byte *sentence, uint32 fontRes, uint8
...
+			Common::String textString = Common::convertBiDiString(textLogical, Common::kWindows1255);
+			currTxtLine  = reinterpret_cast<const byte *>(textString.c_str());
+		}
 		// copy the sprite for each character in this line to the
 		// text sprite and inc the sprite ptr by the character's
 		// width minus the 'overlap'
 
 		for (uint j = 0; j < line[i].length; j++) {
-			byte *charPtr = findChar(sentence[pos++], charSet);
+			byte *charPtr = findChar(*currTxtLine++, charSet);
```
Comment (bluegr) [inline engines/sword2/maketext.cpp:285]: pos is no longer updated here

File: engines/sword2/maketext.cpp [hunk 2]
```diff
@@ -293,7 +299,7 @@ byte *FontRenderer::buildTextSprite(const byte *sentence, uint32 fontRes, uint8
 		}
 
 		// Skip space at end of last word in this line
-		pos++;
+		sentence += line[i].length + 1;
```
Comment (bluegr) [inline engines/sword2/maketext.cpp:302]: pos is no longer updated here

File: engines/sword2/maketext.cpp (hunk 1 above)
Comment (orgads) [reply engines/sword2/maketext.cpp:285]: Yeah, it's no longer used. Removed it.

File: engines/sword2/maketext.cpp (hunk 2 above)
Comment (orgads) [reply engines/sword2/maketext.cpp:302]: ditto

**Comment by orgads** [comment]
Done, and fixed also the window caption.

**Comment by ShaharAriel** [comment]
There is approval from me (author of main changes) to merge.

**Comment by orgads** [comment]
@bluegr Ready for review

**Comment by bluegr** [comment]
Looks good now, thanks! Merging

**Pull request merged**